pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+                pd += self.evaluate_obs(obs, unshifted_args, kwargs)
+
+            # restore the original parameter
+            op.params[p_idx] = orig
+
+        return pd
+
+    def _pd_parameter_shift_var(self, idx, args, kwargs):
+        """Partial derivative of the variance of an observable using the analytic method.
+
+        Args:
+            idx (int): flattened index of the parameter wrt. which the p.d. is computed
+            args (array[float]): flattened positional arguments at which to evaluate the p.d.
+            kwargs (dict[str, Any]): auxiliary arguments
+
+        Returns:
+            array[float]: partial derivative of the node
+        """
+        # boolean mask: elements are True where the return type is a variance, False for expectations
+        where_var = [e.return_type is ObservableReturnTypes.Variance for e in self.circuit.observables]
+        var_observables = [e for e in self.circuit.observables if e.return_type == ObservableReturnTypes.Variance]
+
+        # first, replace each var(A) with <A^2>
+        new_observables = []
+        for e in var_observables:
+            # need to calculate d<A^2>/dp
+            w = e.wires
+
+            if self.model == 'qubit':
+                if e.name == 'Hermitian':
+                    # since arbitrary Hermitian observables
+                    # are not guaranteed to be involutory, need to take them into
+                    # account separately to calculate d<A^2>/dp
+
+                    A = e.params[0]  # Hermitian matrix
+                    #if not np.allclose(A @ A, np.identity(A.shape[0])):
+                    new = qml.expval(qml.Hermitian(A @ A, w, do_queue=False))
+                else:
+                    # involutory, A^2 = I
+                    # For involutory observables (A^2 = I) we have d<A^2>/dp = 0
+                    new = qml.expval(qml.Hermitian(np.identity(2**len(w)), w, do_queue=False))
+
+            else:  # CV circuit, first order observable
+                # get the heisenberg representation
+                # This will be a real 1D vector representing the
+                # first order observable in the basis [I, x, p]
+                A = e._heisenberg_rep(e.parameters) # pylint: disable=protected-access
+
+                # take the outer product of the heisenberg representation
+                # with itself, to get a square symmetric matrix representing
+                # the square of the observable
+                A = np.outer(A, A)
+                new = qml.expval(qml.ops.PolyXP(A, w, do_queue=False))
+
+            # replace the var(A) observable with <A^2>
+            self.circuit.update_node(e, new)
+            new_observables.append(new)
+
+        # calculate the analytic derivatives of the <A^2> observables
+        pdA2 = self._pd_parameter_shift(idx, args, kwargs, force_order2=(self.model == 'cv'))  # FIXME the force_order2 use here is convoluted and could be better
```

Comment:
This was `self.type` in the previous PR, right? Has it changed to `self.model`, or is this a different attribute?

Reply:
Same thing, renamed.